Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.h

```diff
@@ -236,5 +318,130 @@ namespace Terrain
         AZStd::map<AZ::EntityId, TerrainAreaData, TerrainLayerPriorityComparator> m_registeredAreas;
 
         mutable TerrainRaycastContext m_terrainRaycastContext;
+
+        AZ::JobManager* m_terrainJobManager = nullptr;
     };
+
+    template<typename SynchronousFunctionType, typename VectorType>
+    inline AZStd::shared_ptr<AzFramework::Terrain::TerrainDataRequests::TerrainJobContext> TerrainSystem::ProcessFromListAsync(
+        SynchronousFunctionType synchronousFunction,
+        const AZStd::span<VectorType>& inPositions,
+        AzFramework::Terrain::SurfacePointListFillCallback perPositionCallback,
+        Sampler sampleFilter,
+        AZStd::shared_ptr<ProcessAsyncParams> params) const
+    {
+        // Determine the number of jobs to split the work into based on:
+        // 1. The number of available worker threads.
+        // 2. The desired number of jobs as passed in.
+        // 3. The number of positions being processed.
+        const int32_t numWorkerThreads = m_terrainJobManager->GetNumWorkerThreads();
+        const int32_t numJobsDesired = params ? params->m_desiredNumberOfJobs : ProcessAsyncParams::NumJobsDefault;
+        const int32_t numJobsMax = (numJobsDesired > 0) ? AZStd::min(numWorkerThreads, numJobsDesired) : numWorkerThreads;
+        const int32_t numPositionsToProcess = static_cast<int32_t>(inPositions.size());
+        const int32_t numJobs = AZStd::min(numJobsMax, numPositionsToProcess);
```

Comment:
We may want to be even stricter here. If numJobsMax was 8 and there were 8 positions, it'd still be silly to spawn 8 jobs for that work. I think it'd be better to have a MinPositionsPerJob constant set somewhere and make sure each job has at least that much work to do (except in cases where there's only one job of course)

Reply:
Yeah good call, I thought of doing something like that at some point, but I was getting confused enough as it was so I shelved that plan for later once we fix the mutex contention issues to make it viable to split the work into multiple jobs in the first place (I think/hope that might actually already be taken care of with the other change to use the bulk APIs: https://github.com/o3de/o3de/pull/7357).

Reply:
Looking at this in isolation I wasn't confused anymore, so added it:)